hal/common/critical.c

```diff
@@ -236,7 +232,14 @@ bool core_util_atomic_cas_u32(uint32_t *ptr, uint32_t *expectedCurrentValue, uin
     return success;
 }
 
-uint8_t core_util_atomic_incr_u8(uint8_t * valuePtr, uint8_t delta)
+bool core_util_atomic_cas(void **ptr, void **expectedCurrentValue, void *desiredValue) {
```

Comment:
same as in the incr() and decr(), by using an un-suffixed name, you may have taken possession of a generic name.